storage/key_index.go

```diff
@@ -221,3 +240,20 @@ func (g *generation) walk(f func(rev reversion) bool) int {
 func (g *generation) String() string {
 	return fmt.Sprintf("g: ver[%d], revs %#v\n", g.ver, g.revs)
 }
+
+func (a generation) equal(b generation) bool {
+	if a.ver != b.ver {
+		return false
```

Comment:
Hrm, I am a bit concerned about mixing the naming of "version" and "revision". I assume "ver" is supposed to symbolize the "latest revision"?